src/ol/format/wktformat.js

```diff
@@ -491,12 +491,23 @@ ol.format.WKT.Lexer.prototype.nextToken = function() {
 ol.format.WKT.Lexer.prototype.readNumber_ = function() {
   var c, index = this.index_;
   var decimal = false;
+  var scientificNotation = false;
   do {
     if (c == '.') {
       decimal = true;
+    } else if (c !== undefined && c.toLowerCase() == 'e') {
```

Comment:
See https://github.com/openlayers/ol3/blob/master/src/ol/format/wktformat.js#L513 (`toUpperCase`): maybe the character is always `E` ? (haven't tried)